### PR TITLE
  patrons: restore fee count and refactoring patron history

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Fixes count on fees tab and total on header.
* Uses the `operation logs` inex to display the patron history.
* Closes rero/rero-ils#2319.
* Closes rero/rero-ils#1902.
* Closes rero/rero-ils#1227.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Dependencies

* https://github.com/rero/rero-ils/pull/2456

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/circulation/patron/history/history-log/history-log.component.html [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+<div class="col" *ngVar="log.metadata.loan.item.document.pid | getRecord:'documents':'object':'':{ Accept:'application/rero+json' } | async as document">
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/history/history-log/history-log.component.html:17]: is `class="col"` is necessary ?

File: projects/admin/src/app/circulation/patron/history/history.component.html [hunk 2]
```diff
@@ -14,26 +14,21 @@
...
+    <div *ngFor="let log of logs">
+      <admin-history-log [log]="log" class="row mb-1 p-1 border rounded"></admin-history-log>
     </div>
-  </div>
+  </ng-container>
 </section>
 
 <ng-template #noHistory>
   {{ 'No history.' | translate }}
 </ng-template>
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/history/history.component.html:34]: Could you please place this block just below the `ng-container` statement (into the `section` block)

File: projects/public-search/src/app/patron-profile/patron-profile-histories/patron-profile-history/patron-profile-history.component.html [hunk 3]
```diff
@@ -14,24 +14,16 @@
...
-      aria-controls="collapseEvent"
-    >
-      <i class="fa pr-0" [ngClass]="{ 'fa-caret-right': isCollapsed, 'fa-caret-down': !isCollapsed }"></i>
-    </button>
-    <a href="/{{ viewcode }}/documents/{{ record.metadata.document.pid }}">{{ record.metadata.document.title | mainTitle }}</a>
+    <a href="/{{ viewcode }}/documents/{{ record.metadata.loan.item.document.pid }}">
+      {{ record.metadata.loan.item.document.title }}
+    </a>
   </div>
   <div class="col-lg-4 col-sm-4">
```
Comment (zannkukai) [inline projects/public-search/src/app/patron-profile/patron-profile-histories/patron-profile-history/patron-profile-history.component.html:23]: just use `col-4`

File: projects/public-search/src/app/patron-profile/patron-profile-histories/patron-profile-history/patron-profile-history.component.html [hunk 4]
```diff
@@ -43,51 +35,12 @@
         <ng-container *ngIf="!last">; </ng-container>
       </li>
     </ul>
+    </ng-container>
+    <ng-template #norecord>
+      &nbsp;
+    </ng-template>
   </div>
   <div class="col-lg-2 col-sm-2">
```
Comment (zannkukai) [inline projects/public-search/src/app/patron-profile/patron-profile-histories/patron-profile-history/patron-profile-history.component.html:43]: col-2

File: projects/public-search/src/app/patron-profile/patron-profile-ill-requests/patron-profile-ill-request/patron-profile-ill-request.component.html [hunk 5]
```diff
@@ -47,7 +47,7 @@
 
 <!-- Detail -->
 <section id="ill-detail-{{ record.metadata.pid }}" [collapse]="isCollapsed" [isAnimated]="true" class="col-12">
-  <dl class="row">
+  <dl class="row pl-3">
```
Comment (zannkukai) [inline projects/public-search/src/app/patron-profile/patron-profile-ill-requests/patron-profile-ill-request/patron-profile-ill-request.component.html:50]: why ?

**Review by zannkukai — APPROVED** [review]

**Comment by pronguen** [comment]
- [ ] **[*really* not blocking]** When a history contains a loan linked to a deleted item: ok. If the document has been deleted: the display is fine in the public view. In the professional interface, the concerned line cannot be extended.
    image

File: projects/public-search/src/app/patron-profile/patron-profile-ill-requests/patron-profile-ill-request/patron-profile-ill-request.component.html (hunk 5 above)
Comment (Garfield-fr) [reply projects/public-search/src/app/patron-profile/patron-profile-ill-requests/patron-profile-ill-request/patron-profile-ill-request.component.html:50]: To correctly align the description data

File: projects/admin/src/app/circulation/patron/main/main.component.ts [hunk 6]
```diff
@@ -146,6 +153,9 @@ export class MainComponent implements OnInit, OnDestroy {
         // null and this will cause error for navigation url construction
         this._unregisterShortcuts();
         this._registerShortcuts();
+        this._operationLogsApiService.getCheckInHistory(patron.pid, 1, 1).subscribe((result: any) => {
+          this.historyCount = this._recordService.totalHits(result.hits.total);
+        });
         this._patronService.getCirculationInformations(patron.pid).subscribe((data) => {
           this.feesTotalAmount = data.fees.engaged + data.fees.preview;
           this._parseStatistics(data.statistics || {});
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/main/main.component.ts:161]: Is that the history statistics could come with `getCirculationInformations` api call ? With this code we send a second API call to get specific stat. Maybe the best way will be to change the way of compute the 'history' value of the `getCirculationInformations` backend API ?

**Review by zannkukai — APPROVED** [review]

**Review by benerken — APPROVED** [review]

**Pull request merged**